### PR TITLE
Added support for DateTime, DateTimeOffset and TimeSpan

### PR DESCRIPTION
**Pull request by GooRiOn**
# Info
This Pull Request is related to issue no. 46 ( #46 ) 

# Changes
For DateTime and DateTimeOffset types I added:
- IsAfter
- IsBefore
- IsAfterOrEqualTo
- IsBeforeOrEqualTo
- IsAfterNow
- IsBeforeNow
- IsAfterUtcNow
- IsBeforeUtcNow
- IsAfterOrEqualToNow
- IsBeforeOrEqualToNow
- IsAfterOrEqualToUtcNow
- IsBeforeOrEqualToUtcNow
- IsEqualTo
- Required

For TimeSpan type I added:
- IsLessThan
- IsGreaterThan
- IsLessThanOrEqalTo
- IsGreaterThanOrEqualTo
- IsEqualTo
- IsNonZero
- Required

All type extensions support Nullable<T>.

### CODE REVIEW

**Comment by GooRiOn** [comment]
Something is weird happening with AppVeyor... I removed that but still requires status check.... great

**Comment by GooRiOn** [comment]
Ok, fixed that

**Comment by dbarwikowski** [comment]
Is it confirmed we need all 4 combinations of nullable?

File: src/Valit/Rules/Extensions/ValitRuleDoubleExtensions.cs [hunk 1]
```diff
@@ -4,7 +4,7 @@ namespace Valit
 {
     public static class ValitRuleDoubleExtensions
     {
-        public static IValitRule<TObject, double> IsGreaterThan<TObject>(this IValitRule<TObject, double> rule, double value) where TObject : class
+        public static IValitRule<TObject, double> IsGreaterThan<TObject>(this IValitRule<TObject, double> rule, double value, double epislon = 0.0d) where TObject : class
```
Comment (dbarwikowski) [inline src/Valit/Rules/Extensions/ValitRuleDoubleExtensions.cs:7]: Typo. Shouldn't be set to something more than 0?

File: src/Valit/Rules/Extensions/ValitRuleDateTimeOffsetExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,229 @@
+using System;
+
+namespace Valit
+{
+    public static class ValitRuleDateTimeOffsetOffsetExtensions
```
Comment (dbarwikowski) [inline src/Valit/Rules/Extensions/ValitRuleDateTimeOffsetExtensions.cs:5]: OffsetOffset?

**Review by dbarwikowski — CHANGES_REQUESTED** [review]

**Review by dbarwikowski — APPROVED** [review]

File: src/Valit/Rules/Extensions/ValitRuleDoubleExtensions.cs (hunk 1 above)
Comment (arekbal) [reply src/Valit/Rules/Extensions/ValitRuleDoubleExtensions.cs:7]: It is our default which means... ignore epsilon for now, unless you wanna set it explicitly for something like Double.Epsilon. ;)

**Pull request merged**